src/pytest/test.py

```diff
@@ -1508,6 +1492,53 @@ def testNonDefaultDb(self):
         with self.assertResponseError():
             self.cmd('FT.CREATE', 'idx2', 'schema', 'txt', 'text')
 
+    def testDuplicateNonspecFields(self):
+        self.cmd('FT.CREATE', 'idx', 'schema', 'txt', 'text')
+        self.cmd('FT.ADD', 'idx', 'doc', 1.0, 'fields',
+            'f1', 'f1val', 'f1', 'f1val2', 'F1', 'f1Val3')
+
+        res = self.cmd('ft.get', 'idx', 'doc')
+        res = {res[i]: res[i + 1] for i in range(0, len(res), 2)}
+        self.assertTrue(res['f1'] in ('f1val', 'f1val2'))
+        self.assertEqual('f1Val3', res['F1'])
+
+    def testDuplicateFields(self):
+        self.cmd('FT.CREATE', 'idx', 'SCHEMA', 'txt', 'TEXT', 'num', 'NUMERIC', 'SORTABLE')
+        for _ in self.retry_with_reload():
+            # Ensure the index assignment is correct after an rdb load
+            with self.assertResponseError():
+                self.cmd('FT.ADD', 'idx', 'doc', 1.0, 'FIELDS',
+                        'txt', 'foo', 'txt', 'bar', 'txt', 'baz')
+
+            # Try add hash
+            self.cmd('HMSET', 'newDoc', 'txt', 'foo', 'Txt', 'bar', 'txT', 'baz')
+            # Get the actual value:
+
+            from redis import ResponseError
```

Comment:
I didn't want to use assertResponseError to ensure the error is actually because of dup and not because of something else. We should actually fix rmtest to allow us to inspect the specific kind of error received.